tests/test_asyncio/test_cwe_404.py

```diff
@@ -0,0 +1,126 @@
+import asyncio
+import sys
+
+import pytest
+
+from redis.asyncio import Redis
+from redis.asyncio.cluster import RedisCluster
+
+
+async def pipe(
+    reader: asyncio.StreamReader, writer: asyncio.StreamWriter, delay: float, name=""
+):
+    while data := await reader.read(1000):
```

Comment:
The walrus operator was only introduced in Python 3.8 (see [PEP 572](https://peps.python.org/pep-0572/)).

Since this library supports Python 3.7 as well, this line should be changed to:

```suggestion
    while True:
        data = await reader.read(1000)
        if not data:
            break
```

Reply:
Thank you :D